main.tf

```diff
@@ -159,6 +159,14 @@ resource "azurerm_kubernetes_cluster" "main" {
       subnet_id    = var.ingress_application_gateway_subnet_id
     }
   }
+  dynamic "key_management_service" {
+    for_each = var.kms_enabled ? ["key_management_service"] : []
```

Comment:
KMS etcd encryption doesn't work with system-assigned managed identity. You should check for `var.kms_enabled` and for `var.identity_type` to be `UserAssigned`.

Reply:
Done